security-analysis/security-analysis-api/src/main/java/com/powsybl/security/distributed/SecurityAnalysisExecutionHandler.java

```diff
@@ -74,12 +74,31 @@
         void customizeOptions(Path workinDir, SecurityAnalysisCommandOptions options);
     }
 
+    /**
+     * Defines the creation of computation exceptions, in particular log reading.
+     */
+    @FunctionalInterface
+    interface ExceptionHandler {
+        ComputationException createComputationException(Path workinDir, Exception cause);
+    }
+
+    /**
+     * Creates a new security analysis execution handler.
+     *
+     * @param reader Defines how results should be read from working directory.
+     * @param optionsCustomizer If not {@code null}, defines additional command options.
+     * @param exceptionHandler If not {@code null}, used to translate exceptions to a {@link ComputationException}.
+     * @param executionCount The number of executions of the command.
+     * @param input The execution input data.
+     */
     public SecurityAnalysisExecutionHandler(ResultReader<R> reader,
                                             OptionsCustomizer optionsCustomizer,
+                                            ExceptionHandler exceptionHandler,
                                             int executionCount,
                                             SecurityAnalysisExecutionInput input) {
         this.reader = requireNonNull(reader);
         this.optionsCustomizer = optionsCustomizer;
+        this.exceptionHandler = exceptionHandler;
```

Comment:
Is null value OK? (same question for the customizer)

Reply:
I will remove the null possibility : it's correctly handled but not used.